src/app/pages/psbt-request/components/psbt-decoded-request/psbt-decoded-request-nodes/outputs/psbt-decoded-outputs.tsx

```diff
@@ -0,0 +1,59 @@
+import * as btc from '@scure/btc-signer';
+import { Box, Text } from '@stacks/ui';
+
+import { satToBtc } from '@app/common/money/unit-conversion';
+
+import { PsbtBtcOutput } from '../psbt-btc-output';
+import { PsbtDecodedOutputLayout } from './psbt-decoded-output.layout';
+import { PsbtPlaceholderOutputs } from './psbt-placeholder-outputs';
+
+// TODO: Handle all address types?
+function getAddressFromScript(script: Uint8Array) {
+  const outScript = btc.OutScript.decode(script);
+
+  if (outScript.type === 'wsh') return btc.programToWitness(0, script.slice(2));
+  if (outScript.type === 'wpkh') return btc.programToWitness(0, script.slice(2));
+  if (outScript.type === 'tr') return btc.programToWitness(1, script.slice(2));
+  return '';
+}
```

Comment:
It'd be great to have this logic tested. Think this could also better belong in shared crypto folder.

Also, would make 3 separate fns for this and merge together for the "catch all" use. What if I wanna run on only a single type